apps/src/lib/util/firehose.js

```diff
@@ -211,6 +212,8 @@ class FirehoseClient {
       return;
     }
 
+    validateFirehoseDataSize(data.data);
```

Comment:
It seemed redundant to catch an error here just to throw it again, so leaving simple.

Reply:
since we are no longer raising an error in this method, you also need to skip the putRecord call if validation failed.

Reply:
Jinx! I just saw this comment. Okay with the solution I went with?

Reply:
yes, the new solution looks good!